File: client/src/app/modules/tiers/services/billing.closure.recipient.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { BillingClosureRecipientType } from '../model/BillingClosureRecipientType';

@Injectable({
  providedIn: 'root'
})
export class BillingClosureRecipientTypeService extends AppRestService<BillingClosureRecipientType>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getBillingClosureRecipientTypes() {
    return this.getListCached(new HttpParams(), "billing-closure-recipient-types");
  }

  addOrUpdateBillingClosureRecipientType(billingClosureRecipientType: BillingClosureRecipientType) {
    this.clearListCache(new HttpParams(), "billing-closure-recipient-types");
    return this.addOrUpdate(new HttpParams(), "billing-closure-recipient-type", billingClosureRecipientType, "Enregistré", "Erreur lors de l'enregistrement");
  }
}

File: src/main/java/com/jss/osiris/modules/osiris/tiers/service/BillingClosureRecipientTypeService.java
package com.jss.osiris.modules.osiris.tiers.service;

import java.util.List;

import com.jss.osiris.modules.osiris.tiers.model.BillingClosureRecipientType;

public interface BillingClosureRecipientTypeService {
    public List<BillingClosureRecipientType> getBillingClosureRecipientTypes();

    public BillingClosureRecipientType getBillingClosureRecipientType(Integer id);

    public BillingClosureRecipientType addOrUpdateBillingClosureRecipientType(
            BillingClosureRecipientType billingClosureRecipientType);
}
